### ISSUE TITLE
Localization using ES6 features leads to minify failure

### ISSUE THREAD

**Issue by afelicioni** [date-fns/date-fns#1796]
Hello, using `date-fns` for an existing project built via https://create-react-app.dev/

At build stage I am facing a failure as described on https://create-react-app.dev/docs/troubleshooting/#npm-run-build-fails-to-minify since used `react-script` version in project is less than 2.

Current affected version is `v2.14.0` that merged code from #1765 two days ago; my temporary solution is to freeze to `v2.13.0` package version, but since just one file raises the failure, it could be just written in a legacy flavor.

**Comment by afelicioni** [comment]
Just to provide an hint, the file is [src/locale/hu/_lib/formatDistance/index.js](https://github.com/date-fns/date-fns/tree/master/src/locale/hu/_lib/formatDistance/index.js); minification would be successful with the following code variant
```
var translations = {
  about: 'körülbelül',
  over: 'több mint',
  almost: 'majdnem',
  lessthan: 'kevesebb mint'
};
var withoutSuffixes = {
  xseconds: ' másodperc',
  halfaminute: 'fél perc',
  xminutes: ' perc',
  xhours: ' óra',
  xdays: ' nap',
  xweeks: ' hét',
  xmonths: ' hónap',
  xyears: ' év'
};
var withSuffixes = {
  xseconds: {
    '-1': ' másodperccel ezelőtt',
    '1': ' másodperc múlva',
    '0': ' másodperce'
  },
  halfaminute: {
    '-1': 'fél perccel ezelőtt',
    '1': 'fél perc múlva',
    '0': 'fél perce'
  },
  xminutes: {
    '-1': ' perccel ezelőtt',
    '1': ' perc múlva',
    '0': ' perce'
  },
  xhours: {
    '-1': ' órával ezelőtt',
    '1': ' óra múlva',
    '0': ' órája'
  },
  xdays: {
    '-1': ' nappal ezelőtt',
    '1': ' nap múlva',
    '0': ' napja'
  },
  xweeks: {
    '-1': ' héttel ezelőtt',
    '1': ' hét múlva',
    '0': ' hete'
  },
  xmonths: {
    '-1': ' hónappal ezelőtt',
    '1': ' hónap múlva',
    '0': ' hónapja'
  },
  xyears: {
    '-1': ' évvel ezelőtt',
    '1': ' év múlva',
    '0': ' éve'
  }
};

function translate(number, addSuffix, key, comparison) {
  var translated = addSuffix ? withSuffixes[key][String(comparison)] : withoutSuffixes[key];

  if (key === 'halfaminute') {
    return translated;
  }

  return number + translated;
}

export default function formatDistance(token, count, options) {
  options = options || {};
  var adverb = token.match(/about|over|almost|lessthan/i);
  var unit = token.replace(adverb, '');
  var result;
  result = translate(count, options.addSuffix, unit.toLowerCase(), options.comparison);

  if (adverb) {
    result = translations[adverb[0].toLowerCase()] + ' ' + result;
  }

  return result;
}

```

**Comment by JMartelot** [comment]
Hi, I'm facing same issue using last version of `date-fns` with IE11.
Seems that negative indices in array are not well support.

Fixing my date-fns dependency to v2.13.0 solved the problem.

**Comment by lolu-decerno-gbg** [comment]
@JMartelot same issue for me, was about to poly-fill but the downgrade to v2.13.0 solved my problems as well.

**Comment by vbraun** [comment]
Also breaks uglify-js

The `{ [-1]: "foo" }` syntax is definitely ES6, thats rather cutting edge for a basic library. I looked but didn't find any clear documentation for what JS spec is required by date-fns.

**Comment by afelicioni** [comment]
since this issue has been solved in released `v2.15.0`, it hasn't any reason to be open anymore
